bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/binding/math/internal/profiles/MathTransformationProfileFactory.java

```diff
@@ -0,0 +1,160 @@
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.i18n.LocalizedKey;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.thing.profiles.Profile;
+import org.openhab.core.thing.profiles.ProfileCallback;
+import org.openhab.core.thing.profiles.ProfileContext;
+import org.openhab.core.thing.profiles.ProfileFactory;
+import org.openhab.core.thing.profiles.ProfileType;
+import org.openhab.core.thing.profiles.ProfileTypeBuilder;
+import org.openhab.core.thing.profiles.ProfileTypeProvider;
+import org.openhab.core.thing.profiles.ProfileTypeUID;
+import org.openhab.core.thing.profiles.i18n.ProfileTypeI18nLocalizationService;
+import org.openhab.core.transform.TransformationService;
+import org.openhab.core.util.BundleResolver;
+import org.osgi.framework.Bundle;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link MathTransformationProfileFactory} is responsible for creating profiles.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { ProfileFactory.class, ProfileTypeProvider.class })
+public class MathTransformationProfileFactory implements ProfileFactory, ProfileTypeProvider {
+
+    private static final ProfileType ADD_PROFILE_TYPE = ProfileTypeBuilder
+            .newState(AddTransformationProfile.PROFILE_TYPE_UID, AddTransformationProfile.PROFILE_TYPE_UID.getId())
+            .withSupportedItemTypes(CoreItemFactory.NUMBER).build();
+    private static final ProfileType MULTIPLY_PROFILE_TYPE = ProfileTypeBuilder
+            .newState(MultiplyTransformationProfile.PROFILE_TYPE_UID,
+                    MultiplyTransformationProfile.PROFILE_TYPE_UID.getId())
+            .withSupportedItemTypes(CoreItemFactory.NUMBER).build();
+    private static final ProfileType DIVIDE_PROFILE_TYPE = ProfileTypeBuilder
+            .newState(DivideTransformationProfile.PROFILE_TYPE_UID,
+                    DivideTransformationProfile.PROFILE_TYPE_UID.getId())
+            .withSupportedItemTypes(CoreItemFactory.NUMBER).build();
+    private static final Set<ProfileTypeUID> SUPPORTED_PROFILE_TYPE_UIDS = Set.of(
+            AddTransformationProfile.PROFILE_TYPE_UID, MultiplyTransformationProfile.PROFILE_TYPE_UID,
+            DivideTransformationProfile.PROFILE_TYPE_UID);
+    private static final Stream<ProfileType> SUPPORTED_PROFILE_TYPES = Stream.of(ADD_PROFILE_TYPE,
+            MULTIPLY_PROFILE_TYPE, DIVIDE_PROFILE_TYPE);
+
+    private final Map<LocalizedKey, ProfileType> localizedProfileTypeCache = new ConcurrentHashMap<>();
+
+    private final ProfileTypeI18nLocalizationService profileTypeI18nLocalizationService;
+    private final Bundle bundle;
+
+    @NonNullByDefault({})
+    private TransformationService addTransformationService;
```

Comment:
```suggestion
    private @NonNullByDefault({}) TransformationService addTransformationService;
```

seems more consistent with the other null-annotations code style